docker-entrypoint.sh

```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+
+go install github.com/githubnemo/CompileDaemon &&
```

Comment:
Is the `go install` command should be changed to `go get`?
This is my error output when I run `docker compose up`:
<img width="689" alt="image" src="https://github.com/DoCentre/DoCentre/assets/52515354/373d227f-e5de-4a5c-8149-dba26a0f3cb9">

Reply:
After changing the `go install` to `go get`, I get the following errors: 🥹
<img width="814" alt="image" src="https://github.com/DoCentre/DoCentre/assets/52515354/d23fa674-e4c9-4573-a302-9abe2b386dee">

Reply:
The first issue is due to the missing version after the package name, which is addressed by correcting the command to `go install github.com/githubnemo/CompileDaemon@latest`.
We're not using `go get` here because it will add _CompileDaemon_ as a dependency of our module (you can see that _go.mod_ and _go.sum_ are affected). However, since it's a development tool, we would like to install it to _$HOME/go/bin_ with `go install`.

Reply:
The first issue is fixed with rebasing; I'll address the second one. :ok_man:

Reply:
> After changing the `go install` to `go get`, I get the following errors: 🥹 <img alt="image" width="814" src="https://private-user-images.githubusercontent.com/52515354/324234203-d23fa674-e4c9-4573-a302-9abe2b386dee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3MDQ4ODgsIm5iZiI6MTcxMzcwNDU4OCwicGF0aCI6Ii81MjUxNTM1NC8zMjQyMzQyMDMtZDIzZmE2NzQtZTRjOS00NTczLWEzMDItOWFiZTJiMzg2ZGVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDIxVDEzMDMwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI2ZjU3YzJjNTQ3YTMxZjY1NzQ1NDY4NWNmMTJlOGNhZTgwZDgzMmFiODg2YTUzZWNmNGRmMDA2NGEzY2NjY2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ntA_5Bn0Ag-f0vICWzkGByxrsBqJEZl4LAquwosYWxw">

The output message seems unusual because we're using the privileges of the _user_ (1000:1000) in the container, and `go` should attempt to store the cache at _$HOME/.cache_ instead of at the root directory.

Also, I cannot reproduce the situation. :sob: 
@lohsuan Could you enter the container and look around to inspect the current privileges?

You can enter the container without stopping the compose using the following command in another terminal or session:

```bash
docker exec -it <container> /bin/bash
```

Reply:
I also got the following errors after `docker compose up`

![Screenshot from 2024-04-23 13-00-58](https://github.com/DoCentre/DoCentre/assets/100898232/f5cef20f-58f7-4e5b-813c-e17bacdd9d2d)

Reply:
Does this show the privileges of the user in container?

![Screenshot from 2024-04-23 13-05-30](https://github.com/DoCentre/DoCentre/assets/100898232/14cb0a4d-51b3-4d2b-b00b-13118a8d1705)

Reply:
> Does this show the privileges of the user in container?
> 
> ![Screenshot from 2024-04-23 13-05-30](https://private-user-images.githubusercontent.com/100898232/324693664-14cb0a4d-51b3-4d2b-b00b-13118a8d1705.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM4NTA0NTksIm5iZiI6MTcxMzg1MDE1OSwicGF0aCI6Ii8xMDA4OTgyMzIvMzI0NjkzNjY0LTE0Y2IwYTRkLTUxYjMtNGQyYi1iMDBiLTEzMTE4YThkMTcwNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyM1QwNTI5MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kY2MyMTE0N2I0ZDE4MDZkYzJjNmRhZDZkMjdhZDY4N2NlYmZkYzVjMDNmOGFlMzIxMGFkYTJjNmE1M2NmMjhhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ckdxLoMtgApRN6XIg2qbrYRpMI5WWd_tCjLcYX098yE)

This command shows the privileges under the _database_ container. However, the error is about the _server_ container, which has exited. Changing the execution of _CompileDaemon_ in _docker-entrypointer.sh_ to something like `cat` or `sleep infinity` without any argument can keep the container stuck and thus keep it alive.

Reply:
Yes, the database is ran under the _root_ privilege.

Reply:
This should be it
![Screenshot from 2024-04-23 13-45-41](https://github.com/DoCentre/DoCentre/assets/100898232/52fc10c4-3a9d-4305-9497-aca6d97bf580)